src/it/scala/uk/gov/hmrc/govukfrontend/support/TemplateIntegrationSpec.scala

```diff
@@ -4,27 +4,32 @@ import org.jsoup.Jsoup
 import org.scalacheck.Prop.{forAll, secure}
 import org.scalacheck.{Arbitrary, Properties, ShrinkLowPriority, Test}
 import org.scalatest.concurrent.{IntegrationPatience, ScalaFutures}
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
 import play.api.libs.json.{Json, OWrites}
 import uk.gov.hmrc.govukfrontend.support.Implicits._
 import uk.gov.hmrc.govukfrontend.support.ScalaCheckUtils.{ClassifyParams, classify}
 import uk.gov.hmrc.govukfrontend.views.TemplateDiff._
 import uk.gov.hmrc.govukfrontend.views.{JsoupHelpers, PreProcessor, TemplateValidationException, TwirlRenderer}
 import scala.util.{Failure, Success}
+import scala.reflect.ClassTag
 
 /**
   * Base class for integration testing a Twirl template against the Nunjucks template rendering service
   *
   * @tparam T Type representing the input parameters of the Twirl template
   */
-abstract class TemplateIntegrationSpec[T: OWrites: Arbitrary](govukComponentName: String, seed: Option[String] = None)
-    extends Properties(govukComponentName)
+abstract class TemplateIntegrationSpec[T: OWrites: Arbitrary, C: ClassTag](
```

Comment:
Would this be easier to read if the class just took in a type C and then called `classTag[C]`, or am I overthinking?

Reply:
I think @natcarey had to add the type class to satisfy the compiler.